src/header-generator.js

```diff
@@ -0,0 +1,310 @@
+const fs = require('fs');
+const path = require('path');
+const { BayesianNetwork } = require('bayesian-network');
+const { default: ow } = require('ow');
+
+const headerNetworkDefinitionPath = path.join(__dirname, './data_files/header-network-definition.json');
+const inputNetworkDefinitionPath = path.join(__dirname, './data_files/input-network-definition.json');
+const browserHelperFilePath = path.join(__dirname, './data_files/browser-helper-file.json');
+const headersOrderFilePath = path.join(__dirname, './data_files/headers-order.json');
```

Comment:
See comments on line 144 and 156. It would be better to open those files immediately and once, instead of doing it repeatedly in the constructor. You can simply `require` a JSON and you will get the parsed JSON.

```js
const headerNetworkDefinition = require('./data_files/header-network-definition.json'));
headerNetworkDefinition.foo = 'bar';
```